e2e/tests/room/09-jumptomessage.spec.js

```diff
@@ -0,0 +1,295 @@
+			}
+		}
+		await device.launchApp({ permissions: { notifications: 'YES' }, delete: true });
+		await navigateToLogin();
+		await login(data.users.regular.username, data.users.regular.password);
+		await navigateToRoom(threadRoom);
+	});
+
+	describe('Load messages on chat', async() => {
+		it('should jump to an old message and load its surroundings', async() => {
+			await waitFor(element(by.label('Sample message for thread'))).toExist().withTimeout(5000);
+			await sleep(5000); 
+			// we should change this once we come up with a fix
+			await waitFor(element(by.label('Load More')).atIndex(0)).toExist().withTimeout(8000);
+			await element(by.label('Load More')).atIndex(0).tap();
+			await waitFor(element(by.label('Load More')).atIndex(0)).toExist().withTimeout(8000);
+			await element(by.label('Load More')).atIndex(0).tap();
+			await waitFor(element(by.label('Load More')).atIndex(0)).toExist().withTimeout(8000);
+			await element(by.label('Load More')).atIndex(0).tap();
+			await waitFor(element(by.label('Load More')).atIndex(0)).toExist().withTimeout(8000);
+			await element(by.label('Load More')).atIndex(0).tap();
+			await waitFor(element(by.label('Reply')).atIndex(0)).toExist().withTimeout(5000);
+			await element(by.label('Reply')).atIndex(0).tap();
+			await waitFor(element(by.label('200'))).toExist().withTimeout(5000);
+			await waitFor(element(by.label('30'))).toExist().withTimeout(5000);
+			await element(by.label('30')).tap();
+			await waitFor(element(by.label('29'))).toExist().withTimeout(8000);
+			await waitFor(element(by.label('31'))).toExist().withTimeout(8000);
+			await sleep(3000);
+		});
+
+		it('should load messages on scroll', async() => {
+			await waitFor(element(by.id('nav-jump-to-bottom'))).toExist().withTimeout(5000);
+			await element(by.id('nav-jump-to-bottom')).tap();
+			await waitFor(element(by.id('room-view'))).toExist().withTimeout(5000);
+			await sleep(5000);
+			await element(by.id('room-view-messages')).atIndex(1).swipe('down', 'fast', 0.8);
+			await waitFor(element(by.label('160'))).toExist().withTimeout(5000);
+		});
+
+		it('should look for old message and load its surroundings', async() => {
+			await tapBack();
+			await element(by.id('room-view-search')).tap();
+			await waitFor(element(by.id('search-messages-view'))).toExist().withTimeout(5000);
+			await element(by.id('search-message-view-input')).typeText('30\n');
+			await sleep(1000);
+			await waitFor(element(by.label('30')).atIndex(0)).toExist().withTimeout(8000);
+			await element(by.label('30')).atIndex(1).tap();
+			await waitFor(element(by.label('30'))).toExist().withTimeout(10000);
+			await waitFor(element(by.label('32'))).toExist().withTimeout(6000);
+			await waitFor(element(by.label('31'))).toExist().withTimeout(6000);
+			await waitFor(element(by.label('29'))).toExist().withTimeout(6000);
+			await waitFor(element(by.label('28'))).toExist().withTimeout(6000);
+		})
+	});
+
+	describe('Jump to message on threads', async() => {
```

Comment:
- It should tap on thread on main channel and go to the thread
- It should tap on quote
- It should jump from search
- Add a comment for threads pagination
- It should jump to a thread from another room